tracer/test/test-applications/integrations/Samples.StackExchange.Redis/Program.cs

```diff
@@ -13,15 +13,33 @@ class Program
         // was occasionally EXPIRE.
         private static readonly TimeSpan ExpireTimeSpan = TimeSpan.FromMilliseconds(1500);
 
-        static void Main(string[] args)
+        static int Main(string[] args)
         {
             string prefix = "";
             if (args.Length > 0)
             {
                 prefix = args[0];
             }
 
-            RunStackExchange(prefix);
+            try
+            {
+                RunStackExchange(prefix);
+                return 0;
+            }
+            catch (Exception ex)
+                when (ex.GetType().Name == "RedisConnectionException"
+                   && ex.Message.Contains("No connection is available to service this operation"))
+            {
+                // If the redis server is being too slow in responding, we can end up with timeouts
+                // We could do retries, but then we risk butting up against timeout limits etc
+                // As a workaround, we use the specific exit code 13 to indicate a faulty program,
+                // and skip the test.
+                // We need to keep the catch very specific here, so that we don't accidentally
+                // start skipping tests when we shouldn't be
+                Console.WriteLine("Unexpected exception during execution " + ex);
+                Console.WriteLine("Exiting with skip code (13)");
+                return 13;
+            }
         }
```

Comment:
Should we add a delay in this application while we're here?

```csharp
#if NETCOREAPP2_1
            // Add a delay to avoid a race condition on shutdown: https://github.com/dotnet/coreclr/pull/22712
            // This would cause a segmentation fault on .net core 2.x
            System.Threading.Thread.Sleep(5000);
#endif
```

Reply:
Good call 👍

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/4549/commits/eac9c3bd81f16f2caa6bfbb8082d8fd2b49c1e57